x-pack/platform/plugins/shared/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -76,28 +89,113 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
       ...('projectId' in this.config ? { 'OpenAI-Project': this.config.projectId } : {}),
     };
 
-    this.openAI =
-      this.config.apiProvider === OpenAiProviderType.AzureAi
-        ? new OpenAI({
-            apiKey: this.secrets.apiKey,
-            baseURL: this.config.apiUrl,
-            defaultQuery: { 'api-version': getAzureApiVersionParameter(this.config.apiUrl) },
-            defaultHeaders: {
-              ...this.headers,
-              'api-key': this.secrets.apiKey,
-            },
-          })
-        : new OpenAI({
-            baseURL: removeEndpointFromUrl(this.config.apiUrl),
-            apiKey: this.secrets.apiKey,
-            defaultHeaders: {
-              ...this.headers,
-            },
+    // Assign configAny for dynamic property access
+    this.configAny = this.config as Config & {
```

Comment:
No typecasting here. Please use `this.config` as I've recommended (`certificateFile in this.config` checks)

Reply:
this is not resolved. please remove `configAny`